charmcraft/main.py

```diff
@@ -345,6 +353,10 @@ def main(argv=None):
         setup_parts()
         dispatcher = Dispatcher(argv[1:], COMMAND_GROUPS)
         retcode = dispatcher.run()
+    except ArgumentParsingError as err:
+        print(err)
```

Comment:
I'd suggest printing to stderr instead.

Reply:
yes! good catch

Reply:
improved

Reply:
Good catch

Reply:
Also added a test for this, as I found it wasn't covered.